TinREAD.js

```diff
@@ -0,0 +1,277 @@
+{
+	"translatorID": "78835a5b-3378-49c2-a94f-3422aab0e949",
+	"label": "TinREAD",
+	"creator": "Franklin Pezzuti Dyer",
+	"target": "^https?://.+/opac/bibliographic_view",
+	"minVersion": "5.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2024-01-10 03:27:11"
+}
+
+/*
+    ***** BEGIN LICENSE BLOCK *****
+
+    Copyright © 2024 Franklin Pezzuti Dyer
+
+    This file is part of Zotero.
+
+    Zotero is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Zotero is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+    ***** END LICENSE BLOCK *****
+*/
+
+const ICON_DETECT_MAPPING = {
+	"s_book.gif": "journalArticle",
+	"s_e_resource.gif": "journalArticle",
+	"book.gif": "book"
+};
+
+function detectWeb(doc, _url) {
+	// Rather than parsing text in two different languages, we are using the icon
+	let typeIcon = doc.getElementsByClassName("crs_recordtype_icon")[0].src;
+	let iname;
+	for (iname in ICON_DETECT_MAPPING) {
+		if (typeIcon.includes(iname)) {
+			return ICON_DETECT_MAPPING[iname];
+		}
+	}
+	return false;
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('li.reslt_item_head > a[name="book_link"]');
+	for (let row of rows) {
+		let href = row.href;
+		let title = ZU.trimInternal(row.textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+async function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'multiple') {
+		let items = await Zotero.selectItems(getSearchResults(doc, false));
+		if (!items) return;
+		for (let url of Object.keys(items)) {
+			await scrape(await requestDocument(url));
+		}
+	}
+	else {
+		await scrape(doc, url);
+	}
+}
+
+async function scrape(doc, _url = doc.location.href) {
+	doc.getElementById("DirectLink").click();
```

Comment:
We can't rely on direct interaction with the webpage. While that will work in some current environments, it won't work in all environments (e.g., translation-server) and generally won't be possible in browsers in the future. So for a case like this, you'd want to figure out what the click kicked off and replicate that in code.